src/test/java/com/at/avro/integration/HsqlIntegrationTest.java

```diff
@@ -0,0 +1,175 @@
+package com.at.avro.integration;
+
+import com.at.avro.AvroSchema;
+import com.at.avro.DbSchemaExtractor;
+import com.at.avro.SchemaGenerator;
+import com.at.avro.config.AvroConfig;
+import com.at.avro.config.FormatterConfig;
+import com.at.avro.formatters.SchemaFormatter;
+import com.at.avro.mappers.RemovePlural;
+import com.at.avro.mappers.ToCamelCase;
+import org.flywaydb.core.Flyway;
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+import java.sql.Connection;
+import java.sql.DriverManager;
+import java.sql.SQLException;
+import java.util.List;
+import java.util.function.Function;
+
+import static helper.Utils.classPathResourceContent;
+import static org.hamcrest.CoreMatchers.is;
+import static org.junit.Assert.assertThat;
+
+/**
+ * @author artur@callfire.com
+ */
+public class HsqlIntegrationTest {
+
+    private static final String CONNECTION_URL = "jdbc:hsqldb:mem:testcase;shutdown=true";
+
+    private AvroConfig avroConfig = new AvroConfig("test.namespace");
+
+    static private DbSchemaExtractor SCHEMA_EXTRACTOR;
+
+    private static Connection CONNECTION;
```

Comment:
Looks like there's no need to make these static. Maybe just change @BeforeClass to @Before?

Reply:
Those fields are made static to improve test speed. We don't need to create connection and extractor object before each tests. Instead we are creating them only once and then reuse.

Reply:
Can you check if that makes a noticeable difference? I prefer to not have static fields where they are not needed.

Also per java convention order of filed modifiers should be "private static final" here and in PgsqlIntegrationTest - can you please fix?

Reply:
The whole flyway oriented refactoring is based in the assumption that DB is created once per suite, migrations are applied once per suite and then the DB connection used for the former things is used for creating an instance of `com.at.avro.DbSchemaExtractor` that will be reused across tests.

This design is imposed by the fact the DB populatino and JDBC conncetion creation are time consuming operations and should be avoided when possible.
Integration tests are perfectly fitting into this, since there's no need of recreating the connection on each run.

> Also per java convention order of filed modifiers should be "private static final" here and in PgsqlIntegrationTest - can you please fix?
This only applies for what is called _constants_. Unfortunately due to limitaitons of java and junit it's quite hard to achieve this. I have to reimplement junit rules to initialize final static vars on junit rule creation.

Reply:
It's possible to do a Flyway migration once per test class (in @BeforeClass) and not reuse DbSchemaExtractor and Connection (create new in @Before). I don't think it will make a difference in test execution time or memory consumption. Even though it's not critical I'd still like to avoid static state.


> This only applies for what is called constants ...

I'm not saying we need to add "final" everywhere :) I'm saying that field modifiers should be in the same order everywhere and we can use the order suggested by Java code convention.  So "static private" should be "private static", etc.

Reply:
Even if we don't examine visible difference on execution time - the idea is that time consuming operations should be avoided and not executed for each tests case. For instance in future in may lead to serious degradation as soon as this project start to have more integration tests cases. In this connection creation is a quite time \ resource consuming operation. 

After reviving `DbSchemaExtractor` I tend to agree that this should be non static field.

Also I'd like to raise a question that seems as a sort of design flaw, i.e. the fact the extractor creates a connection each time it retrieves schema from DB. Instead extractor should be provided with DataSource object, so connection could be retrieved from downstream connection pool or JNDI or whatever. But this is definitely out of scope.

Reply:
Updated PR

Reply:
Totally agree, creating connections by hand in `DbSchemaExtractor` is not the best choice and should be refactored. Thanks for bringing it up!

I still think that it's ok to create a separate connection for each test and don't bother with reusing it. It's not that big a deal though, so +1.